client/src/modules/stock/inventory-adjustment/inventory-adjustment.js

```diff
@@ -0,0 +1,273 @@
+    enableColumnMenus : false,
+    columnDefs : columns,
+    data : vm.Stock.store.data,
+    fastWatch : true,
+    flatEntityAccess : true,
+    rowTemplate : 'modules/templates/grid/error.row.html',
+    onRegisterApi : onRegisterApiFn,
+  };
+
+  // register api
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  // inline filter
+  vm.toggleInlineFilter = () => {
+    vm.gridOptions.enableFiltering = !vm.gridOptions.enableFiltering;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  };
+
+  // on lot select
+  function onLotSelect(row) {
+    if (!row.lot || !row.lot.uuid) { return; }
+
+    checkValidity();
+    refreshSelectedLotsList();
+  }
+
+  // update the list of selected lots
+  function refreshSelectedLotsList() {
+    vm.selectedLots = vm.Stock.store.data
+      .filter(item => item.lot && item.lot.uuid)
+      .map(item => item.lot.uuid);
+  }
+
+  // add items
+  function addItems(n) {
+    vm.Stock.addItems(n);
+    checkValidity();
+  }
+
+  // remove item
+  function removeItem(item) {
+    vm.Stock.removeItem(item.id);
+    checkValidity();
+  }
+
+  // configure item
+  function configureItem(item) {
+    item._initialised = true;
+    // get lots
+    Stock.lots.read(null, { depot_uuid : vm.depot.uuid, inventory_uuid : item.inventory.inventory_uuid })
+      .then((lots) => {
+        item.lots = lots.filter(lot => !vm.selectedLots.includes(lot.uuid));
+      })
+      .catch(Notify.handleError);
+  }
+
+  function setupStock() {
+    vm.selectedLots = [];
+    vm.Stock.setup();
+    vm.Stock.store.clear();
+  }
+
+  function startup() {
+    vm.movement = {
+      date : new Date(),
+      entity : {},
+    };
+  }
+
+  // ============================ Inventories ==========================
+  function loadInventories(depot) {
+    setupStock();
+
+    Stock.inventories.read(null, { depot_uuid : depot.uuid })
+      .then((inventories) => {
+        vm.selectableInventories = angular.copy(inventories);
+        return Stock.lots.read(null, {
+          depot_uuid : depot.uuid,
+          includeEmptyLot : 0,
+          dateTo : vm.movement.date,
+        });
+      })
+      .then(lots => {
+        addItems(lots.length);
+        vm.Stock.store.data.forEach((item, index) => {
+          const lot = lots[index];
+          item.inventory = {
+            inventory_uuid : lot.inventory_uuid,
+            text : lot.text,
+            code : lot.code,
+          };
+          item.lot = lot;
+          item.quantity = lot.quantity;
+          // added for sorting on these columns
+          item.code = lot.code;
+          item.description = lot.text;
+          item.label = lot.label;
+          // set item lots details
+          configureItem(item);
+        });
+        checkValidity();
+      })
+      .catch(Notify.handleError);
+  }
+
+  // check validity
+  function checkValidity() {
+    const lotsExists = vm.Stock.store.data.every((item) => {
```

Comment:
It looks like this is used in multiple modules.  I think we should probably put this code in a method in the [`StockFormService`](https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/modules/stock/StockForm.service.js#L1) so that we can call `vm.Stock.isValid()` or something.